File: harvey/9012.py
def vps(ps):
    stack = []
    for s in range(len(ps)):
        if ps[s]=='(':
            stack.append(1)
        elif ps[s]==')':
            if stack:
                stack.pop()
            else:
                return 'NO'
    return 'NO' if stack else 'YES'


n = int(input())

for i in range(n):
    print(vps(input()))

File: chu/solved/9012.java
package week2;

import java.util.*;
public class aaa9012 {
	public static String vps(String pm){
		Stack st = new Stack();
		for(int i = 0; i < pm.length(); ++i){
			if(pm.charAt(i) == '(')
				st.push(1);
			else if(pm.charAt(i) == ')')
				if(st.isEmpty())
					return "NO";
				else
					st.pop();
		}
		if(st.isEmpty())
			return "YES";
		else
			return "NO";
	}
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int tc = sc.nextInt();
		
		for(int i = 0; i < tc; ++i){
			String stdin = sc.next();
			System.out.println(vps(stdin));
		}
	}
}
